rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -2948,111 +2958,152 @@ void spu_thread::do_putlluc(const spu_mfc_cmd& args)
 	vm::reservation_notifier(addr).notify_all(-128);
 }
 
-void spu_thread::do_mfc(bool /*wait*/)
+void spu_thread::do_mfc(bool can_escape)
 {
 	u32 removed = 0;
 	u32 barrier = 0;
 	u32 fence = 0;
 
-	// Process enqueued commands
-	static_cast<void>(std::remove_if(mfc_queue + 0, mfc_queue + mfc_size, [&](spu_mfc_cmd& args)
+	auto get_exec_mask = [&size = mfc_size]
 	{
-		// Select tag bit in the tag mask or the stall mask
-		const u32 mask = utils::rol32(1, args.tag);
+		return static_cast<u16>((0 - (1u << std::min<u32>(g_cfg.core.mfc_transfers_shuffling, size))) | __rdtsc());
+	};
+
+	// Shuffle commands execution (if enabled), explicit barriers are obeyed
+	u16 exec_mask = get_exec_mask();
+	bool pending = false;
 
-		if ((args.cmd & ~0xc) == MFC_BARRIER_CMD)
+	// Process enqueued commands
+	while (true)
+	{
+		static_cast<void>(std::remove_if(mfc_queue + 0, mfc_queue + mfc_size, [&](spu_mfc_cmd& args)
```

Comment:
Can you put remove_if lambda into a variable?

Reply:
Fixed.